.github/workflows/lint.yml

```diff
@@ -3,6 +3,31 @@ name: Lint
 on: [push]
 
 jobs:
+  codeql:
+    name: Github CodeQL
+    runs-on: ubuntu-20.04
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby' ]
+        language: [ 'go', 'python' ]
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: ${{ matrix.language }}
+        # this covers a larger portion of checks but are experimental right now
+        queries: security-and-quality
+    - run: make build
```

Comment:
Why do we need to build here?

Reply:
@askolesov forgot to update this PR, I thought I already did. added codeql to the test.yml file

Reply:
I mean can we run codeql in a separate job that doesn't build our sources?

Reply:
we can do it without building but we'll just be missing a check. CodeQL can analyse build process and the output binary so we should definitely use it